bpx/schema.py

```diff
@@ -279,20 +301,50 @@ class Parameterisation(ExtraBaseModel):
     electrolyte: Electrolyte = Field(
         alias="Electrolyte",
     )
-    negative_electrode: Electrode = Field(
+    negative_electrode: Union[ElectrodeSingle, ElectrodeBlended] = Field(
         alias="Negative electrode",
     )
-    positive_electrode: Electrode = Field(
+    positive_electrode: Union[ElectrodeSingle, ElectrodeBlended] = Field(
         alias="Positive electrode",
     )
     separator: Contact = Field(
         alias="Separator",
     )
 
 
+class ParameterisationSPM(ExtraBaseModel):
+    cell: Cell = Field(
+        alias="Cell",
+    )
+    negative_electrode: Union[ElectrodeSingleSPM, ElectrodeBlendedSPM] = Field(
+        alias="Negative electrode",
+    )
+    positive_electrode: Union[ElectrodeSingleSPM, ElectrodeBlendedSPM] = Field(
+        alias="Positive electrode",
+    )
+
+
 class BPX(ExtraBaseModel):
     header: Header = Field(
         alias="Header",
     )
-    parameterisation: Parameterisation = Field(alias="Parameterisation")
+    parameterisation: Union[ParameterisationSPM, Parameterisation] = Field(
+        alias="Parameterisation"
+    )
     validation: Dict[str, Experiment] = Field(None, alias="Validation")
+
+    @root_validator(skip_on_failure=True)
+    def model_based_validation(cls, values):
+        model = values.get("header").model
+        parameter_class_name = values.get("parameterisation").__class__.__name__
+        allowed_combinations = [
+            ("Parameterisation", "DFN"),
+            ("Parameterisation", "SPMe"),
+            ("ParameterisationSPM", "SPM"),
+        ]
+        if (parameter_class_name, model) in allowed_combinations:
+            return values
+        else:
+            raise ValueError(
```

Comment:
can we raise a warning instead of an error? think error might be aggressive and will end up being a barrier to https://github.com/pybamm-team/BPX/issues/40

Reply:
Is there any native way to raise a warning in `pydantic`?
It looks like it hasn't been implemented (see this [issue](https://github.com/pydantic/pydantic/issues/939)).

Reply:
Changed it to a warning. Updated tests and CHANGELOG.
Please feel free to suggest a better warning message if it sounds awkward :)